### PR TITLE
Update F-OH-DATA

### PR DESCRIPTION
**Pull request by imansmallapple**
1. Translated Chinese Readme file into English
2. Added app deploy to oniro appStore procedure via docker
3. Added math punching app's icon and hap file
4. Added Dockerfile where the deploy procedure used

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,58 +1,160 @@
...
+```typescript
+docker build -t f-oh-data .
+```
+
+Re-run the `f-oh-data` image file  
+```typescript
+docker run -d -p 5000:5000 f-oh-data
+```
+
+problem solved
```
Comment (frankplus) [inline README.md]: these instructions are not relevant for this readme. These are enough.

````suggestion
## Start the server with Docker
### **Build the Docker Image**
Run the following command:

```bash
docker build -t f-oh-data .
```

### **Run the Docker Container**
To start the container:

```bash
docker run -d -p 5000:5000 f-oh-data
```
````

**Review by frankplus — CHANGES_REQUESTED** [review]
- can you update `allAppList.json` adding the math punching app
- please separate the commits: one commit for readme translation, one for hap and app list update, one for docker instruction addition to readme. Also with proper commit descrition

**Comment by imansmallapple** [comment]
Commit seperated

File: README.md (hunk 1 above)
Comment (frankplus) [inline README.md]: These changes should not be in the README translation commit

File: README.md [hunk 2]
```diff
@@ -25,136 +27,57 @@ Or
...
-```typescript
-docker build -t f-oh-data .
-```
-
-Re-run the `f-oh-data` image file  
-```typescript
-docker run -d -p 5000:5000 f-oh-data
-```
-
-problem solved
```
Comment (frankplus) [inline README.md]: after removing it from the "README translation" commit, these part should go away

**Review by frankplus — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (imansmallapple) [reply README.md]: This part was removed from git history now

**Comment by frankplus** [comment]
The commit author should be `chensong64@h-partners.com` for all the commits:

```
commit fcfe8895a012d476348b6b19e24c72218d4923dd (HEAD -> oniro, origin/oniro, origin/HEAD)
Author: chen <alf138540fun@gmail.com>
Date:   Thu Nov 21 20:21:04 2024 +0100

    Docker instruction addition to readme
    
    Signed-off-by: Chen <chensong64@h-partners.com>

commit 519f3349f0e6213bb86045c6f6aacbf85633d03e
Author: chen <alf138540fun@gmail.com>
Date:   Thu Nov 21 20:17:25 2024 +0100

    Add hap file and app list update
    
    Signed-off-by: Chen <chensong64@h-partners.com>

commit 735f5799ed519136262add03037d49624a49ccd2
Author: chen <alf138540fun@gmail.com>
Date:   Thu Nov 21 20:16:15 2024 +0100

    add Dockerfile
    
    Signed-off-by: Chen <chensong64@h-partners.com>

commit 9cac3293a85a02ec23dfca47943992e29165bb8d
Author: Chen <chensong64@h-partners.com>
Date:   Fri Nov 22 11:10:50 2024 +0100

    README translation
    
    Signed-off-by: Chen <chensong64@h-partners.com>
```

**Comment by imansmallapple** [comment]
> The commit author should be `chensong64@h-partners.com` for all the commits:
> 
> ```
> commit fcfe8895a012d476348b6b19e24c72218d4923dd (HEAD -> oniro, origin/oniro, origin/HEAD)
> Author: chen <alf138540fun@gmail.com>
> Date:   Thu Nov 21 20:21:04 2024 +0100
> 
>     Docker instruction addition to readme
>     
>     Signed-off-by: Chen <chensong64@h-partners.com>
> 
> commit 519f3349f0e6213bb86045c6f6aacbf85633d03e
> Author: chen <alf138540fun@gmail.com>
> Date:   Thu Nov 21 20:17:25 2024 +0100
> 
>     Add hap file and app list update
>     
>     Signed-off-by: Chen <chensong64@h-partners.com>
> 
> commit 735f5799ed519136262add03037d49624a49ccd2
> Author: chen <alf138540fun@gmail.com>
> Date:   Thu Nov 21 20:16:15 2024 +0100
> 
>     add Dockerfile
>     
>     Signed-off-by: Chen <chensong64@h-partners.com>
> 
> commit 9cac3293a85a02ec23dfca47943992e29165bb8d
> Author: Chen <chensong64@h-partners.com>
> Date:   Fri Nov 22 11:10:50 2024 +0100
> 
>     README translation
>     
>     Signed-off-by: Chen <chensong64@h-partners.com>
> ```

The author was changed back now.

**Pull request merged**